Classes/Common/DocumentList.php

```diff
@@ -316,6 +252,122 @@ protected function getRecord($element)
         return $record;
     }
 
+    /**
+     * It gets SOLR result
+     *
+     * @access private
+     *
+     * @param array $record: for searched document
+     *
+     * @return Result
+     */
+    private function getSolrResult($record) {
+        $fields = Solr::getFields();
+
+        $query = $this->solr->service->createSelect();
+        // Restrict the fields to the required ones
+        $query->setFields($fields['uid'] .',' . $fields['id'] .',' . $fields['toplevel'] .',' . $fields['thumbnail'] .',' . $fields['page']);
+        foreach ($this->solrConfig as $solr_name) {
+            $query->addField($solr_name);
+        }
+        // Set additional query parameters.
+        // Set reasonable limit for safety reasons.
+        // We don't expect to get more than 10.000 hits per UID.
+        $query->setStart(0)->setRows(1000);
+        // Take over existing filter queries.
+        $filterQueries = isset($this->metadata['options']['params']['filterquery']) ? $this->metadata['options']['params']['filterquery'] : [];
+        // Extend filter query to get all documents with the same UID.
+        foreach ($filterQueries as $key => $value) {
+            if (isset($value['query'])) {
+                $filterQuery[$key] = $value['query'] . ' OR ' . $fields['toplevel'] . ':true';
+                $query->addFilterQuery($filterQuery);
+            }
+        }
+        // Add filter query to get all documents with the required uid.
+        $filterQuery['key'] = $fields['uid'] . ':' . Solr::escapeQuery($record['uid']);
+        $query->addFilterQuery($filterQuery);
```

Comment:
```suggestion
        $query->createFilterQuery('uid')->setQuery($fields['uid'] . ':' . Solr::escapeQuery($record['uid']));
```
This doesn't work and will be ignored. As result, you always search fulltext:(something) OR toplevel:true and receive ALL toplevel documents. But you only want the toplevel element for the given uid.

`addFilterQuery()` expects an array like this:
`$filterQuery = [ 'key' => 'fq1', 'query' => 'uid:1' ];`

So I'd suggest to use `createFilterQuery()` like in the [documentation of Solarium](https://solarium.readthedocs.io/en/4.2.0/queries/select-query/building-a-select-query/adding-filterqueries/). Maybe you find another solution.